fix(lint): lint fixes

abhi-markan opened the pull request:
## Introduction
Linting errors

## Resolution
* Various lint fixes across all the micro-services.

## Miscellaneous
* Dependency updates

JennyHumphries commented on dtfs-central-api/api-tests/api.js:
```diff
@@ -11,7 +12,9 @@ module.exports = (app) => ({
     to: async (url) => {
       const results = [];
 
+      // eslint-disable-next-line no-restricted-syntax
```
These two rules (Line 15 and 17) are disabled in some of the other services. Should we disable them for this service as well rather than opting out on individual lines? Or is it needed for the non-test files in `dtfs-central-api`?

abhi-markan replied:
Have disabled across the micro-service and removed individual lines from across the files.

JennyHumphries commented on dtfs-central-api/api-tests/wipeDB.js:
```diff
@@ -1,7 +1,8 @@
 const wipe = async (collections) => {
-  const drop = async (collection) => new Promise(async (resolve, reject) => {
+  // eslint-disable-next-line no-async-promise-executor
+  const drop = async (collection) => new Promise(async (resolve) => {
```
I think we can fix this lint error rather than disabling it if we change it to:
```
  const drop = async (collection) => new Promise((resolve) => {
    db.getCollection(collection).then((collection) => {
      collection.drop(() => { resolve() });
    });
  });
```

abhi-markan replied:
Good shout, have refactored.

JennyHumphries commented on portal-api/api-tests/wipeDB.js:
```diff
@@ -1,10 +1,11 @@
 
 const wipe = async (collections) => {
-  const drop = async (collection) => new Promise(async (resolve, reject) => {
+  // eslint-disable-next-line no-async-promise-executor
```
I think this might be able to be fixed in the same way as the other `wipeDB` file

abhi-markan replied:
Have refactored,.

JennyHumphries commented on portal/.eslintrc.js:
```diff
@@ -21,6 +21,7 @@ module.exports = {
     'object-curly-newline': ['error', {
       consistent: true,
     }],
+    'no-restricted-syntax': 0,
```
As in the other config file, please can we change `0` to `'off'`? And on line 18 and 19?

abhi-markan replied:
Have updated.

JennyHumphries commented on utils/data-migration/.eslintrc.js:
```diff
@@ -19,6 +19,10 @@ module.exports = {
+    'import/no-unresolved': 0,
+    'no-await-in-loop': 0,
+    'import/extensions': 0,
+    'no-restricted-syntax': 0,
```
As above, please can we make all the rules use `'off`' instead of 0?

abhi-markan replied:
Have updated.

JennyHumphries commented on utils/data-migration/.eslintrc.js:
```diff
@@ -19,6 +19,10 @@ module.exports = {
     'comma-dangle': 'off',
     'no-loop-func': 'off',
+    'import/no-unresolved': 0,
+    'no-await-in-loop': 0,
```
Now that these rules are disabled here, we should be able to remove all individual instances where it's disabled in this service. For example line 167 in `migrate-deals.js`.

I think there are quite a few places we can make similar changes.

abhi-markan replied:
Have removed individual instances.

JennyHumphries reviewed (changes requested):
Looks good, just a couple of small changes suggested.

abhi-markan commented on portal-api/.eslintrc.js:
```diff
@@ -23,6 +23,8 @@ module.exports = {
     'object-curly-newline': ['error', {
       consistent: true,
     }],
+    'no-restricted-syntax': 0,
```
Have replaced across all the micro-services.

abhi-markan commented:
> Looks good, just a couple of small changes suggested.

Thanks @JennyHumphries can you please re access the PR have addressed all your comments. Thanks

JennyHumphries reviewed (approved):
Looks good to me.

It looks like some of the cypress tests and code checks are failing, I'm not sure if that's something that normally needs to be fixed before merging or not.
